### PR TITLE
post: angular signals in v16

### PR DESCRIPTION
**Pull request by cexbrayat**

### CODE REVIEW

File: _posts/2023-04-25-angular-signals.md [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+    count.set(2);
+    // logs only 2, not 0 and 1, as this is the value when the under-the-hood effect runs
+
+To convert an observable to a signal, we can use the `fromObservable` method:
+
+    const count = fromObservable(count$);
+
+The signal will contain the last value emitted by the observable,
+or will throw an error if the observable emits an error.
+Note that this observable is automatically destroyed when the component that declared it is destroyed.
```
Comment (jnizet) [inline _posts/2023-04-25-angular-signals.md:315]: Unclear. I guess you mean that the subscription created by `fromObservable()` is automatically unsubscribed.

File: _posts/2023-04-25-angular-signals.md [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+
+If you do not provide an initial value, the signal will throw an error
+if it is read before the observable emits a value.
+
+
+## Signal-based components
+
+In the future (v17? v18?),
+we'll be able to build a component with fields that are signals and computed values used in the template.
+The framework would be notified when an expression has changed thanks to the signals, and would thus only need to re-render the components affected by the change, without having to check for changes on unrelated components, without the need for zone.js.
```
Comment (jnizet) [inline _posts/2023-04-25-angular-signals.md:328]: Unclear: we can already do most of that in v16. Maybe you could tell instead that inputs and queries will be signals.

File: _posts/2023-04-25-angular-signals.md [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+
+Effects can also return a cleanup function, that is run when the effect runs again.
+This can be handy when you need to stop a previous action before starting a new one.
+In the example below, we start an interval that runs every `count` milliseconds,
+and we want to stop it and start a new one when the count changes:
+
+    this.intervalEffect = effect(
+      () => {
+        const intervalId = setInterval(() => console.log(`count in intervalEffect ${this.count()}`), this.count());
+        return () => clearInterval(intervalId)
```
[the remark this answers is not in the document]
Comment (cexbrayat) [reply _posts/2023-04-25-angular-signals.md:278]: It did, I need to update 👍

File: _posts/2023-04-25-angular-signals.md (hunk 1 above)
Comment (cexbrayat) [reply _posts/2023-04-25-angular-signals.md:315]: Yes

**Pull request closed without merging**